build_runner/test/common/test_phases.dart

```diff
@@ -124,21 +125,23 @@ Future<BuildResult> testBuilders(
       enableLowResourcesMode: enableLowResourcesMode,
       verbose: verbose,
       buildDirs: buildDirs);
-  var build = await BuildRunner.create(
-    options,
-    environment,
-    builders,
-    builderConfigOverrides,
-    isReleaseBuild: false,
-  );
+
   BuildResult result;
-  if (build == null) {
-    result = new BuildResult(BuildStatus.failure, []);
-  } else {
+  try {
+    var build = await BuildRunner.create(
+      options,
+      environment,
+      builders,
+      builderConfigOverrides,
+      isReleaseBuild: false,
+    );
     result = await build.run({});
     await build.beforeExit();
     await options.logListener.cancel();
+  } on CannotBuildException {
```

Comment:
do we need to catch this here? Do any of our tests hit this behavior? If so, are they worthwhile tests?